### PR TITLE
Migrate platform_design to null safety

### LINKED ISSUE #726: [platform_design] Migrate to null-safety
The [flutter_lorem](https://pub.dev/packages/flutter_lorem) package that `platform_design` is using just got migrated to null-safety. The package author was very quick to review my PR and released a new version on pub.dev just now.

I have started with the migration of the sample. A new PR will be up soon.

/cc @domesticmouse  @RedBrogdon

### PR DESCRIPTION
**Pull request by abd99**
* Upgraded dependencies
* Migrated sample to null-safety
 
closes #726

/cc @filiph @domesticmouse @RedBrogdon

### CODE REVIEW

File: platform_design/lib/widgets.dart [hunk 1]
```diff
@@ -94,7 +92,7 @@ class _PressableCardState extends State<PressableCard>
         behavior: HitTestBehavior.opaque,
         onTap: () {
           if (widget.onPressed != null) {
-            widget.onPressed();
+            widget.onPressed!();
```
Comment (RedBrogdon) [inline platform_design/lib/widgets.dart:95]: It's possible to replace this with `widget.onPressed?.call()` and take out the if-statement, but this way is better for readability.

File: platform_design/lib/widgets.dart [hunk 2]
```diff
@@ -57,8 +55,8 @@ class PressableCard extends StatefulWidget {
 class _PressableCardState extends State<PressableCard>
     with SingleTickerProviderStateMixin {
   bool pressed = false;
-  AnimationController controller;
-  Animation<double> elevationAnimation;
+  late AnimationController controller;
+  late Animation<double> elevationAnimation;
```
Comment (RedBrogdon) [inline platform_design/lib/widgets.dart:59]: `late final`?

File: platform_design/lib/widgets.dart [hunk 3]
```diff
@@ -300,7 +302,7 @@ void showChoices(BuildContext context, List<String> choices) {
                       value: index,
                       groupValue: selectedRadio,
                       // ignore: avoid_types_on_closure_parameters
-                      onChanged: (int value) {
+                      onChanged: (int? value) {
```
Comment (RedBrogdon) [inline platform_design/lib/widgets.dart:305]: You should be able to remove this type specifier entirely (and the comment on line 304), if you set the `RadioListTile` to be a.`RadioListTile<int?>`.

File: platform_design/pubspec.yaml [hunk 4]
```diff
@@ -3,20 +3,20 @@ description: A project showcasing a Flutter app following different platform IA
 version: 1.0.0+1
 
 environment:
-  sdk: ">=2.5.0 <3.0.0"
+  sdk: '>=2.12.0-259.12.beta <3.0.0'
```
Comment (RedBrogdon) [inline platform_design/pubspec.yaml:6]: `2.12.0-0 < 3.0.0` is fine for now.

**Review by RedBrogdon — CHANGES_REQUESTED** [review]
Thanks for the migration, @abd99! I've added a few comments, but nothing serious.

File: platform_design/lib/main.dart [hunk 5]
```diff
@@ -116,7 +116,7 @@ class _PlatformAdaptingHomePageState extends State<PlatformAdaptingHomePage> {
             );
           default:
             assert(false, 'Unexpected tab');
-            return null;
+            return Container();
```
[the remark this answers is not in the document]
Comment (abd99) [reply platform_design/lib/main.dart:119]: I was wondering if there's a better alternative to `Container()` in this case. Thanks!

File: platform_design/lib/news_tab.dart [hunk 6]
```diff
@@ -36,7 +36,7 @@ class _NewsTabState extends State<NewsTab> {
   }
 
   Widget _listBuilder(BuildContext context, int index) {
-    if (index >= _itemsLength) return null;
+    if (index >= _itemsLength) return Container();
```
[the remark this answers is not in the document]
Comment (abd99) [reply platform_design/lib/news_tab.dart:39]: Yup, nice catch!

File: platform_design/lib/news_tab.dart [hunk 7]
```diff
@@ -22,9 +22,9 @@ class NewsTab extends StatefulWidget {
 class _NewsTabState extends State<NewsTab> {
   static const _itemsLength = 20;
 
-  List<Color> colors;
-  List<String> titles;
-  List<String> contents;
+  late List<Color> colors;
```
[the remark this answers is not in the document]
Comment (abd99) [reply platform_design/lib/news_tab.dart:25]: Sure

File: platform_design/lib/widgets.dart [hunk 8]
```diff
@@ -28,7 +26,7 @@ class PlatformWidget extends StatelessWidget {
         return iosBuilder(context);
       default:
         assert(false, 'Unexpected platform $defaultTargetPlatform');
-        return null;
+        return Container();
```
[the remark this answers is not in the document]
Comment (abd99) [reply platform_design/lib/widgets.dart:29]: Okay, so it's more about consistency. Thanks for mentioning it.

File: platform_design/lib/widgets.dart (hunk 1 above)
Comment (abd99) [reply platform_design/lib/widgets.dart:95]: I think we should change it. People will get to learn the use of the `?` operator more, like I did.

**Comment by abd99** [comment]
PTAL @RedBrogdon, I have made the changes. =)

File: platform_design/lib/widgets.dart (hunk 8 above)
[the remark this answers is not in the document]
Comment (RedBrogdon) [reply platform_design/lib/widgets.dart]: Yeah, it should be a tiny bit faster than Container, but the odds of someone ever noticing are infinitesimal.

**Review by RedBrogdon — APPROVED** [review]

**Pull request merged**